prow/scripts/provision-vm-and-start-cluster-users-k3d.sh

```diff
@@ -0,0 +1,118 @@
+    exit $ARG
+}
+
+function testCustomImage() {
+    CUSTOM_IMAGE="$1"
+    IMAGE_EXISTS=$(gcloud compute images list --filter "name:${CUSTOM_IMAGE}" | tail -n +2 | awk '{print $1}')
+    if [[ -z "$IMAGE_EXISTS" ]]; then
+        log::error "${CUSTOM_IMAGE} is invalid, it is not available in GCP images list, the script will terminate ..." && exit 1
+    fi
+}
+
+gcp::authenticate \
+    -c "${GOOGLE_APPLICATION_CREDENTIALS}"
+
+RANDOM_ID=$(openssl rand -hex 4)
+
+LABELS=""
+if [[ -z "${PULL_NUMBER}" ]]; then
+    LABELS=(--labels "branch=$PULL_BASE_REF,job-name=kyma-integration")
+else
+    LABELS=(--labels "pull-number=$PULL_NUMBER,job-name=kyma-integration")
+fi
+
+POSITIONAL=()
+while [[ $# -gt 0 ]]
+do
+
+    key="$1"
+
+    case ${key} in
+        --image)
+            IMAGE="$2"
+            testCustomImage "${IMAGE}"
+            shift
+            shift
+            ;;
+        --*)
+            echo "Unknown flag ${1}"
+            exit 1
+            ;;
+        *)    # unknown option
+            POSITIONAL+=("$1") # save it in an array for later
+            shift # past argument
+            ;;
+    esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+
+if [[ -z "$IMAGE" ]]; then
+    log::info "Provisioning vm using the latest default custom image ..."
+    
+    IMAGE=$(gcloud compute images list --sort-by "~creationTimestamp" \
+         --filter "family:custom images AND labels.default:yes" --limit=1 | tail -n +2 | awk '{print $1}')
+    
+    if [[ -z "$IMAGE" ]]; then
+       log::error "There are no default custom images, the script will exit ..." && exit 1
+    fi   
+ fi
+
+date
+
+ZONE_LIMIT=${ZONE_LIMIT:-5}
+EU_ZONES=$(gcloud compute zones list --filter="name~europe" --limit="${ZONE_LIMIT}" | tail -n +2 | awk '{print $1}')
+STARTTIME=$(date +%s)
+for ZONE in ${EU_ZONES}; do
+    log::info "Attempting to create a new instance named kyma-integration-test-${RANDOM_ID} in zone ${ZONE} using image ${IMAGE}"
+    gcloud compute instances create "kyma-integration-test-${RANDOM_ID}" \
+        --metadata enable-oslogin=TRUE \
+        --image "${IMAGE}" \
+        --machine-type n1-standard-4 \
+        --zone "${ZONE}" \
+        --boot-disk-size 200 "${LABELS[@]}" &&\
+    log::info "Created kyma-integration-test-${RANDOM_ID} in zone ${ZONE}" && break
+    log::error "Could not create machine in zone ${ZONE}"
+done || exit 1
+ENDTIME=$(date +%s)
+echo "VM creation time: $((ENDTIME - STARTTIME)) seconds."
+
+trap cleanup exit INT
+
+log::info "Copying Kyma to the instance"
+#shellcheck disable=SC2088
+utils::compress_send_to_vm "${ZONE}" "kyma-integration-test-${RANDOM_ID}" "/home/prow/go/src/github.com/kyma-project/kyma" "~/kyma"
+
+log::info "Triggering the installation"
+gcloud compute ssh --quiet --zone="${ZONE}" --command="sudo USE_ALPHA=${USE_ALPHA} bash" --ssh-flag="-o ServerAliveInterval=30" "kyma-integration-test-${RANDOM_ID}" < "${SCRIPT_DIR}/cluster-integration/cluster-users-integration-k3d.sh"
```

Comment:
It looks like the job runs with bash tracing enabled (`-x`), though I can't find the place the flag is used. Please disable it for more readable output in the job logs.

Reply:
it's already disabled in `kyma` repo, the logs run are from previous run. they won;'t be there after merge.